paper-api/src/main/java/org/bukkit/inventory/RecipeChoice.java

```diff
@@ -290,4 +343,14 @@ public String toString() {
         }
         // Paper end - check valid ingredients
     }
+
+    /**
+     * Represents a choice that will be valid only if an item matches the
+     * given predicate.
+     */
+    @NullMarked
+    sealed interface PredicateChoice extends RecipeChoice permits PredicateChoiceImpl {
+
+        Predicate<ItemStack> getPredicate();
```

Comment:
Not sure we have to expose this, since it is a predicate itself. An empty interface is fine. To your point about maybe not needing interfaces, I think it's good to have even an empty one so you can at least know what type of choice it is. otherwise it's a little annoying to figure out.